### ISSUE TITLE
[mlpack] Build error

### ISSUE THREAD

**Issue by Teles1** [microsoft/vcpkg#21579]
**Host Environment**
- OS: Microsoft Windows [Version 10.0.19043.1348]
- Compiler: revision
- 
package: mlpack[core]:x86-windows -> 3.4.1#2
  vcpkg-tool version: 2021-11-19-dd4f5f501990cc475022b1677675c344a91b8192
  vcpkg-scripts version: 653912591 2021-11-21 (32 minutes ago)

**To Reproduce**
`./vcpkg install mlpack:x86-windows`

**Failure logs** 
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/7576942/config-x86-windows-out.log)

**Additional context**
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: C:/vcpkg/buildtrees/mlpack/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\mlpack\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/mlpack/portfile.cmake:31 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)

**Comment by JonLiu1993** [comment]
@Teles1 , I am curious why you can install mlpack:x86-windows, the latest mlpack only supports uwp
````
"supports": "uwp",
````
could you please git pull your vcpkg and try again?

**Comment by genisispig** [comment]
@JonLiu1993 For uwp: #21310
For the error in the given log #20773
For the fix: #20439

**Comment by JonLiu1993** [comment]
@Teles1 ,The error was caused by armadillo, which will be fixed by this PR #20439
````
CMake Error at C:/vcpkg/installed/x86-windows/share/Armadillo/ArmadilloConfig.cmake:12 (include):
  include could not find requested file:

    /share/Armadillo/ArmadilloLibraryDepends.cmake
Call Stack (most recent call first):
  C:/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package)
  CMakeLists.txt:294 (find_package)
````

**Comment by JackBoosY** [comment]
Close this issue via #20439 merged.
